AdminCore.Common/Extensions/DateTimeExtensions.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+
+namespace AdminCore.Common.Extensions
+{
+    public static class DateTimeExtensions
+    {
+        // https://stackoverflow.com/questions/1617049/calculate-the-number-of-business-days-between-two-dates
+        /// <summary>
+        /// Calculates number of business days, taking into account:
+        ///  - weekends (Saturdays and Sundays)
+        ///  - bank holidays in the middle of the week
+        /// </summary>
+        /// <param name="firstDay">First day in the time interval</param>
+        /// <param name="lastDay">Last day in the time interval</param>
+        /// <param name="bankHolidays">List of bank holidays excluding weekends</param>
+        /// <returns>Number of business days during the 'span'</returns>
+        public static int BusinessDaysUntil(this DateTime firstDay, DateTime lastDay, params DateTime[] bankHolidays)
```

Comment:
that's a pretty long function with a lot of magic numbers, consider splitting this up and labelling those numbers with something more meaningful to improve readability.

Reply:
I've changed the magic numbers to be defined by private statics with better naming. I don't think splitting it up into different methods will change how readable it is though.